### PR TITLE
Add test for dropping a fixed attribute and adding it back as var-sized

### PR DESCRIPTION
**Pull request by kounelisagis**
Following https://github.com/TileDB-Inc/TileDB/pull/5321 and https://github.com/TileDB-Inc/TileDB/pull/5362, let's add a TileDB-Py test.

---

[sc-57039]

### CODE REVIEW

**Comment by kounelisagis** [comment]
Note for myself: A larger random data generation is needed.

File: tiledb/tests/test_schema_evolution.py [hunk 1]
```diff
@@ -227,3 +227,36 @@ def test_schema_evolution_extend_check_bad_type():
...
+    se.array_evolve(uri)
+
+    se = tiledb.ArraySchemaEvolution(ctx)
+    newattr = tiledb.Attr("a", dtype="S0", var=True)
+    se.add_attribute(newattr)
+    se.array_evolve(uri)
+
+    with tiledb.open(uri) as A:
+        res = A[:]
+        assert_array_equal(res["a"], data)
```
Comment (ihnorton) [inline tiledb/tests/test_schema_evolution.py:261]: Shouldn't the result be empty here?

Comment (bekadavis9) [reply tiledb/tests/test_schema_evolution.py:261]: +1

Comment (kounelisagis) [reply tiledb/tests/test_schema_evolution.py]: Right! Test was modified.

File: tiledb/tests/test_schema_evolution.py [hunk 2]
```diff
@@ -227,3 +227,58 @@ def test_schema_evolution_extend_check_bad_type():
...
+        assert A.schema.has_attr("a")
+        assert A.schema.attr("a").dtype == "S"
+        assert A.schema.attr("b").dtype == np.int32
+
+    # add new data to the array
+    new_data = np.array(
+        ["tiledb-string-n.{}".format(i) for i in range(1, 11)], dtype="S"
+    )
+    with tiledb.open(uri, "w") as A:
+        A[:] = {"a": new_data, "b": original_data}
```
Comment (bekadavis9) [inline tiledb/tests/test_schema_evolution.py:278]: You don't need to write to `b` again; the original data will still be there. I would also discourage writing to `a` here. This test should validate that `a != original_data`. The expected behavior is that the original (`int`) data is lost and the new `char` data is uninitialized. Checking that each value `== b'\x80'` is sufficient, as long as a comment accompanies the check to explain the behavior.

Comment (bekadavis9) [reply tiledb/tests/test_schema_evolution.py:278]: Alternatively, you could change the second-to-last line of this test to some `assert_array_not_equal(res["a"], original_data)`. 
I don't know what specific check could be used - that's your wheelhouse :)

Comment (ihnorton) [reply tiledb/tests/test_schema_evolution.py:278]: we have to write `b` again because single-attribute writes are not supported.

**Review by ihnorton — APPROVED** [review]
LGTM after addressing the fill value check request from @bekadavis9

**Pull request merged**